From b8826cf46273ed70b0834de2a444e99cbdde63e2 Mon Sep 17 00:00:00 2001
From: Serge Hallyn <serge.hallyn@ubuntu.com>
Date: Thu, 18 Sep 2014 14:04:24 -0500
Subject: [PATCH 1/1] apparmor: use TEMPLATE.qemu for kvm

virDomainVirtTypeToString() returns 'qemu' and 'kvm' separately.
Don't require a separate apparmor profile for both, rather always
look for TEMPLATE.qemu.

Signed-off-by: Serge Hallyn <serge.hallyn@ubuntu.com>
---
 src/security/virt-aa-helper.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/security/virt-aa-helper.c b/src/security/virt-aa-helper.c
index a06ba44..0447248 100644
--- a/src/security/virt-aa-helper.c
+++ b/src/security/virt-aa-helper.c
@@ -341,15 +341,19 @@ create_profile(const char *profile, const char *profile_name,
     int tlen, plen;
     int fd;
     int rc = -1;
+    const char *virttype;
 
     if (virFileExists(profile)) {
         vah_error(NULL, 0, _("profile exists"));
         goto end;
     }
 
+    virttype = virDomainVirtTypeToString(virtType);
+    if (strcmp(virttype, "kvm") == 0)
+	    virttype = "qemu";
 
     if (virAsprintfQuiet(&template, "%s/TEMPLATE.%s", APPARMOR_DIR "/libvirt",
-                         virDomainVirtTypeToString(virtType)) < 0) {
+                         virttype) < 0) {
         vah_error(NULL, 0, _("template name exceeds maximum length"));
         goto end;
     }
-- 
2.1.0

